### ISSUE TITLE
RedHat 6 GLIBC and GLIBCXX error

### ISSUE THREAD

**Issue by mich9463** [nodejs/build#1995]
* **Version**: 12.13
* **Platform**:
```
[linux@WDCTyInt0000304 bin]$ cat /etc/redhat-release
Red Hat Enterprise Linux Server release 6.9 (Santiago)
[linux@WDCTyInt0000304 bin]$ uname -a
Linux WDCTyInt0000304.some-domain.com 2.6.32-696.18.7.el6.x86_64 nodejs/node#1 SMP Thu Dec 28 20:15:47 EST 2017 x86_64 x86_64 x86_64 GNU/Linux
```
* **Subsystem**:

Downloaded and extracted node-v12.13.0-linux-x64.tar.xz from nodejs.org

Attempt to run:
```
[linux@WDCTyInt0000304 bin]$ ./node
./node: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.14' not found (required by ./node)
./node: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by ./node)
./node: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.5' not found (required by ./node)
./node: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by ./node)
./node: /lib64/libc.so.6: version `GLIBC_2.16' not found (required by ./node)
./node: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by ./node)
./node: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by ./node)
```

**Comment by sam-github** [comment]
FYI, supported platforms: https://github.com/nodejs/node/blob/v12.x/BUILDING.md#supported-platforms

Unless you think your platform meets the minimum reqs, in which case something might be wrong, this is expected, we'll close.

**Comment by mich9463** [comment]
Thanks for the supported platforms link.  Closing

**Comment by jlbooker** [comment]
This appears to mean that the Node v12.13+ distributed binaries are permanently broken on RHEL6.

Even with the latest gcc and libgcc packages, I still get the same error message as above.

**Comment by bnoordhuis** [comment]
@jlbooker You also need a new enough glibc; centos 6 ships 2.12, IIRC, but you need 2.17+. Try using devtoolset.

**Comment by rvagg** [comment]
@jlbooker see https://github.com/nodejs/node/blob/v12.x/BUILDING.md#official-binary-platforms-and-toolchains (note the `v12.x` branch in that URL). We build the official binaries on a CentOS / RHEL 7 base, that's our minimum glibc target for the lifespan of 12.x. 10.x were the last that we built with CentOS / RHEL 6. This decision is made by considering the support lifetime of the base distributions. CentOS 6 will be EOL in November of this year, Node.js 12.x will still live on through 2021.

You can compile from source yourself though, just get the latest devtoolset and you should be fine.

**Comment by jlbooker** [comment]
Yes, a realize that RHEL/CentOS 6 will be EOL in a few months. However, this is for a production system that will live on a while longer. Being that it's production, I'm not really interested in getting into custom building Node there, nor pulling in versions of glibc/libgcc that are outside of Redhat's packages (as that risks breaking lots of other things).

I was more just noting/confirming for others that find this error after me that the published binaries for v12.13+ won't run on RHEL 6/CentOS 6, even with all of the latest package updates available from Redhat.
